src/Services/ApiKeyService.php

```
@@ -62,45 +62,119 @@ public function update_keys(): void {
+                }
+                $needsMigration = true;
+            }
+        }
+
+        // Clean up legacy keys if we did any migration
+        if ($needsMigration) {
+            $default_params = $this->cleanup_legacy_keys($default_params);
+        }
+
+        return $default_params;
+    }
```

PHPMD (2.15.0)

The method processCentralSettings() has a Cyclomatic Complexity of 27. The configured cyclomatic complexity threshold is 10. (Code Size Rules)

(CyclomaticComplexity)